libs/src/sdk/services/EntityManager/EntityManager.ts

```diff
@@ -107,8 +101,63 @@ export class EntityManager implements EntityManagerService {
 
     const jsonResponse = await response.json()
 
+    if (!skipConfirmation) {
+      await this.confirmWrite({
+        blockHash: jsonResponse.receipt.blockHash,
+        blockNumber: jsonResponse.receipt.blockNumber,
+        confirmationTimeout
+      })
+    }
+
     return {
       txReceipt: jsonResponse.receipt
     }
   }
+
+  /**
+   * Confirms a write by polling for the block to be indexed by the selected
+   * discovery node
+   */
+  async confirmWrite({
+    blockHash,
+    blockNumber,
+    confirmationTimeout = CONFIRMATION_TIMEOUT
+  }: {
+    blockHash: string
```

Comment:
Since we no longer care about blockhash, do we even have to send it?

Reply:
Good q, let me try without

Reply:
We do in fact still need it unless we wanna change this query https://github.com/AudiusProject/audius-protocol/blob/main/discovery-provider/src/queries/get_block_confirmation.py#L11

Reply:
But we don't check block found, just block passed, right?

Reply:
I guess we probably still need to pass the validation, but I think any string will

Reply:
Oh yeah you're right, it's the validation that's breaking without blockhash. Gonna keep passing it for now and then can maybe do a cleanup